base_vat_optional_vies/tests/test_res_partner.py

```diff
@@ -12,14 +12,19 @@ def setUp(self):
         super(TestResPartner, self).setUp()
         self.company = self.env.user.company_id
         self.company.vat_check_vies = True
-        self.partner = self.env["res.partner"].create({"name": "Test partner"})
+        self.partner = self.env["res.partner"].create(
+            {
+                "name": "Test partner",
+                "country_id": self.env.ref("base.es").id,
+                "company_type": "company",
+            }
+        )
         self.vatnumber_path = "odoo.addons.base_vat.models.res_partner.check_vies"
```

Comment:
```suggestion
        self.partner.onchange_company_type()
        self.vatnumber_path = "odoo.addons.base_vat.models.res_partner.check_vies"
```
should do the trick.